diff --git a/node_modules/puppeteer-extra-plugin-anonymize-ua/index.js b/node_modules/puppeteer-extra-plugin-anonymize-ua/index.js
index 7465257..0ecc8bd 100644
--- a/node_modules/puppeteer-extra-plugin-anonymize-ua/index.js
+++ b/node_modules/puppeteer-extra-plugin-anonymize-ua/index.js
@@ -38,7 +38,7 @@ class Plugin extends PuppeteerExtraPlugin {
     }
   }

-  async onPageCreated(page) {
+  async handler (page) {
     let ua = await page.browser().userAgent()
     if (this.opts.stripHeadless) {
       ua = ua.replace('HeadlessChrome/', 'Chrome/')
@@ -52,6 +52,17 @@ class Plugin extends PuppeteerExtraPlugin {
     this.debug('new ua', ua)
     await page.setUserAgent(ua)
   }
+
+  async onBrowser(browser) {
+    const pages = await browser.pages()
+    for (const page of pages) {
+      await this.handler(page)
+    }
+  }
+
+  async onPageCreated(page) {
+    await this.handler(page)
+  }
 }

 module.exports = function(pluginConfig) {
